src/controllers/commands/builtin/twitch/UpdateChannel.cpp

```diff
@@ -105,10 +141,46 @@ QString setGame(const CommandContext &ctx)
                     channel->addMessage(makeSystemMessage(
                         QString("Updated game to %1").arg(matchedGame.name)));
                 },
-                [channel] {
-                    channel->addMessage(
-                        makeSystemMessage("Game update failed! Are you "
-                                          "missing the required scope?"));
+                [channel](auto error, auto message) {
+                    QString errorMessage = QString("Failed to set game - ");
+
+                    using Error = HelixUpdateChannelError;
+
+                    switch(error)
+                    {
+                        case Error::UserMissingScope: {
+                            errorMessage += "Missing required scope. "
+                                            "Re-login with your "
+                                            "account and try again.";
+                        }
+                        break;
+
+                        case Error::UserNotAuthorized: {
+                            errorMessage += "You must be the broadcaster "
+                                            "to set the game.";
+                        }
+                        break;
+
+                        case Error::Ratelimited: {
+                            errorMessage +=
+                                "You are being ratelimited by Twitch. Try "
+                                "again in a few seconds.";
+                        }
+                        break;
+
+                        case Error::Forwarded: {
+                            errorMessage += message;
+                        }
+                        break;
+
+                        case Error::Unknown:
+                        default: {
+                            errorMessage += "An unknown error has occured.";
+                        }
+                        break;
+                    }
```

Comment:
This is almost identical to the code above and should be refactored into a function (in an anonymous namespace above `chatterino::commands`).